lib/puppet/provider/package/apt.rb

```diff
@@ -130,4 +164,27 @@ def purge
   def install_options
     join_options(@resource[:install_options])
   end
+
+  def insync?(is)
+    # this is called after the generic version matching logic (insync? for the
+    # type), so we only get here if should != is
+
+    #if 'should' is a range and 'is' a debian version we should check if 'should' includes 'is'
+    should = @resource[:ensure]
+    begin
+      should_range = VersionRange.parse(should, DebianVersion)
```

Comment:
same as above, only do this if should is a String

Reply:
the check for string is in DebianVersion/VersionRange
if we do it in provider, then we should remove it from DebianVersion/VersionRange?

Reply:
maybe yeah... it's a bit confusing to get a "cannot parse" message when you have a symbol and do not expect parsing to happen

Reply:
there are two things:
- having better exception message - and this should be fixed 
- where to check *Version/Range input validity

Reply:
I would try to parse the input as a range only if it looks like a range, similar to what @Dorin-Pleava did here: https://github.com/puppetlabs/puppet/pull/8030/files#diff-0df6fa18d947b3857abe3ff1c68dfd44R318

I'm not sure if it's faster but it's easier to understand for me at least.